theseus/theseus_layer.py

```diff
@@ -45,6 +46,39 @@ def forward(
         )
         return values, info
 
+    def compute_samples(
+        self,
+        linear_solver: LinearSolver = None,
+        n_samples: int = 10,
+        temperature: float = 1.0,
+    ) -> torch.Tensor:
+        if linear_solver is None:
+            return None
+        delta = linear_solver.solve()
```

Comment:
Is there a way to avoid solving the linear system again? This should be computed in the last iteration in forward.